lib/vector-core/src/mapping/mod.rs

```diff
@@ -36,7 +36,7 @@ impl Function for Assignment {
                 target.as_mut_log().insert(&self.path, v);
                 Ok(())
             }
-            _ => Err("assignment must be from a value".to_string()),
+            QueryValue::Regex(_) => Err("assignment must be from a value".to_string()),
```

Comment:
Similarly here and below I think `if let Some()` would be clearer that we only care about one type.

Reply:
Reversed in d949fa1